### PR TITLE
105 list flirts for company

### PR DESCRIPTION
**Pull request by SilGosker**

### CODE REVIEW

File: Data/Helpers/BCryptHelper.cs [hunk 1]
```diff
@@ -12,6 +12,7 @@ public static void ConfigureUserPassword(User user, string password)
 
     public static bool ValidatePassword(User user, string password)
     {
+        if (user == null) return false;
```
Comment (HUSKUT) [inline Data/Helpers/BCryptHelper.cs:15]: I could've sworn this was already added?

File: EasyIntern-Backend/Controllers/CompanyController.cs [hunk 2]
```diff
@@ -6,22 +6,47 @@
...
     public class CompanyController : Controller
     {
         private readonly Context _context;
         private const int TakeAmount = 10;
+
         public CompanyController(Context context)
         {
             _context = context;
         }
+        [IsCompany]
```
Comment (HUSKUT) [inline EasyIntern-Backend/Controllers/CompanyController.cs]: Change this to authenticated

File: EasyIntern-Backend/Controllers/FlirtController.cs [hunk 3]
```diff
@@ -33,11 +33,7 @@ public async Task<IActionResult> Index()
                 Student = new
                 {
                     e.Student.Name,
-                    ProfileSettings = e.Student.ProfileSettings.Select(o => new
-                    {
-                        o.Key,
-                        o.Value
-                    })
+                    ProfileSettings = new Dictionary<string, string>(e.Student.ProfileSettings.Select(o => new KeyValuePair<string, string>(o.Key, o.Value)))
```
Comment (HUSKUT) [inline EasyIntern-Backend/Controllers/FlirtController.cs]: ugly ass formatting ngl

File: EasyIntern-Backend/Controllers/CompanyController.cs [hunk 4]
```diff
@@ -6,22 +6,47 @@
...
+        [HttpGet("")]
+        public async Task<IActionResult> Index()
+        {
+            if (User.IsModerator()) //if the user is moderator return all companies
+            {
+                return Json(await _context.Users.Where(e => e.UserType == UserType.Company).ToListAsync());
+            }
+
+            if (!User.IsStudent())//user is company, return all students that have flirted with the users company.
```
Comment (SophiaH67) [inline EasyIntern-Backend/Controllers/CompanyController.cs]: Why are we doing a check for `IsStudent`? There's already an `[IsCompany]` gaurd in place

**Review by SophiaH67 — CHANGES_REQUESTED** [review]

File: EasyIntern-Backend/Controllers/CompanyController.cs (hunk 4 above)
Comment (HUSKUT) [reply EasyIntern-Backend/Controllers/CompanyController.cs]: +1 (or ++, whatever code u prefer)

**Review by SophiaH67 — APPROVED** [review]

**Review by HUSKUT — APPROVED** [review]

**Review by SophiaH67 — APPROVED** [review]

**Review by SophiaH67 — APPROVED** [review]

**Review by SophiaH67 — APPROVED** [review]

**Pull request merged**